### PR TITLE
Specify componet ip

### PR DESCRIPTION
**Pull request by smellthemoon**
/kind feature
#26538

### CODE REVIEW

File: pkg/util/funcutil/func.go [hunk 1]
```diff
@@ -49,6 +49,27 @@ func CheckGrpcReady(ctx context.Context, targetCh chan error) {
...
+	}
+	if err == nil {
+		for _, addr := range addrs {
+			ipaddr, ok := addr.(*net.IPNet)
+			if ipaddr.IP.String() == ip && ok && ipaddr.IP.IsGlobalUnicast() && ipaddr.IP.To4() != nil {
+				return ipaddr.IP.String()
+			}
+		}
+	}
+	panic("specified ip is invalid")
```
Comment (jiaoew1991) [inline pkg/util/funcutil/func.go:69]: Why did copy the code for GetLocalIP again? 🤔

Comment (smellthemoon) [reply pkg/util/funcutil/func.go:69]: here has check the incoming ip is one of the local ip and is valid, GetLocalIP just return the first valid ip.

Comment (jiaoew1991) [reply pkg/util/funcutil/func.go:69]: I don't think it's necessary to check. On one hand, the port binding in grpc will perform a check, and on the other hand, we haven't considered the case of IPv6 here. 🤔

**Pull request merged**